### PR TITLE
[UR][CUDA] Fix race condition in CUDA stream creation

### PR DESCRIPTION
**Pull request by rafbiels**
Fix race condition in CUDA stream creation in the UR CUDA adapter

See https://github.com/oneapi-src/unified-runtime/pull/1984

### CODE REVIEW

File: sycl/cmake/modules/FetchUnifiedRuntime.cmake [hunk 1]
```diff
@@ -116,14 +116,12 @@ if(SYCL_UR_USE_FETCH_CONTENT)
...
 
-  set(UNIFIED_RUNTIME_REPO "https://github.com/oneapi-src/unified-runtime.git")
-  # commit f8336050f43a6529636fd6b2075dfe09961564f5
-  # Merge: 6c98e0e8 b75f2bc2
-  # Author: Omar Ahmed <omar.ahmed@codeplay.com>
-  # Date:   Tue Aug 13 13:07:23 2024 +0100
-  #     Merge pull request #1955 from nrspruit/fix_l0_coverity
-  #     [L0] fix Coverity issues for L0 Adapter
-  set(UNIFIED_RUNTIME_TAG f8336050f43a6529636fd6b2075dfe09961564f5)
+  set(UNIFIED_RUNTIME_REPO "https://github.com/rafbiels/unified-runtime.git")
```
Comment (rodburns) [inline sycl/cmake/modules/FetchUnifiedRuntime.cmake:119]: I think this should be https://github.com/oneapi-src/unified-runtime.git and also the path below needs to be changed.

Comment (rafbiels) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake:119]: this is correct for step 2 of the [UR contribution procedure](https://oneapi-src.github.io/unified-runtime/core/CONTRIB.html#adapter-change-process) and gets changed in step 8

Comment (rodburns) [reply sycl/cmake/modules/FetchUnifiedRuntime.cmake:119]: Apologies, my mistake.

**Comment by omarahmed1111** [comment]
This will include bump for these PRs:
* https://github.com/oneapi-src/unified-runtime/pull/1985
* https://github.com/oneapi-src/unified-runtime/pull/1988
* https://github.com/oneapi-src/unified-runtime/pull/1987
* https://github.com/oneapi-src/unified-runtime/pull/1936
* https://github.com/oneapi-src/unified-runtime/pull/1972
* https://github.com/oneapi-src/unified-runtime/pull/1984

**Review by omarahmed1111 — APPROVED** [review]

**Comment by omarahmed1111** [comment]
@intel/llvm-gatekeepers Please merge, Thanks!

**Pull request merged**